src/ray/object_manager/object_manager.cc

```diff
@@ -90,6 +88,19 @@ void ObjectManager::NotifyDirectoryObjectAdd(const ObjectInfoT &object_info) {
   local_objects_[object_id] = object_info;
   ray::Status status =
       object_directory_->ReportObjectAdded(object_id, client_id_, object_info);
+  // Handle the unfulfilled_push_tasks_ which contains the push request that is not
+  // completed due to unsatisfied local objects.
+  auto iter = unfulfilled_push_tasks_.find(object_id);
+  if (iter != unfulfilled_push_tasks_.end()) {
+    for (auto &pair : iter->second) {
+      main_service_->post(
+          [this, object_id, pair]() { RAY_CHECK_OK(Push(object_id, pair.first)); });
+      if (pair.second != nullptr) {
```

Comment:
When would `pair.second` equal `nullptr`?

EDIT: I see, this happens when the `timeout` equals `0`, right? In that case, I think we should just not add a timer.

Reply:
This happens when `timeout < 0`, we need to keep this record in the map until the object notification calls it.

Reply:
Oh I see, can you add a comment here explaining that?